### PR TITLE
Projectile SIze Item done

### PR DESCRIPTION
**Pull request by BasilChristofis**
For the projectile Size Item, just check the item changes the size of the projectile for each slot.

### CODE REVIEW

**Comment by JarrydWorland** [comment]
I think the implementation of the projectile size needs more than what it is. 

At the moment it's purely cosmetic and doesn't serve much purpose - perhaps these should be made to be invulnerable, except on wall collisions, where you could potentially have it affect two or three targets in a row without destroying on contact with an enemy?

**Comment by LachlanMcCaskill** [comment]
For the projectile size item, we need to set it such that it changes how big the projectile gets each time, with diminishing returns. The first time it should be significant, but then it should be smaller every time after that. I'll see if I can implement this using a stack.

**Comment by LachlanMcCaskill** [comment]
Update: I don't think it can be done using a stack, but I think it can be done by changing the value of addScale based on the current scale of the projectile (provided you already have access)

File: MicrowaveGame/Assets/Resources/Scenes/Gameplay.unity [hunk 1]
```diff
@@ -500,6 +500,10 @@ PrefabInstance:
       propertyPath: m_IsTrigger
       value: 0
       objectReference: {fileID: 0}
+    - target: {fileID: 8573620642782398239, guid: 71a2878986dd1e048b0d34eac7a690b9, type: 3}
+      propertyPath: AimAssistConeAngle
+      value: 0
+      objectReference: {fileID: 0}
```
Comment (102107466) [inline MicrowaveGame/Assets/Resources/Scenes/Gameplay.unity:506]: I don't think this was intentional but it looks like the aim assist cone angle has been set to 0 for the gameplay scene.

File: MicrowaveGame/Assets/Resources/Prefabs/Projectiles/ProjectileBullet.prefab [hunk 2]
```diff
@@ -113,6 +113,10 @@ PrefabInstance:
       propertyPath: m_LocalScale.y
       value: 0.1
       objectReference: {fileID: 0}
+    - target: {fileID: 7178730013344640899, guid: f53cd03f428072b4a9ad1ac029ec8236, type: 3}
+      propertyPath: m_LocalScale.z
+      value: 0
+      objectReference: {fileID: 0}
```
Comment (102107466) [inline MicrowaveGame/Assets/Resources/Prefabs/Projectiles/ProjectileBullet.prefab:119]: Not sure why this was changed but transform scale z should not be 0, I would leave it the same as the other values (0.1) or 1.0

File: MicrowaveGame/Assets/Resources/Scripts/Player/PlayerShootBehaviour.cs [hunk 3]
```diff
@@ -88,6 +88,7 @@ private void Start()
 
 			_projectileSpawnTransform = transform.Find("ProjectileSpawn");
 			_aimIndicatorTransform = transform.Find("AimIndicator");
+			_projectilePrefab.transform.localScale = new Vector3(0.1f, 0.1f, 0);
```
Comment (102107466) [inline MicrowaveGame/Assets/Resources/Scripts/Player/PlayerShootBehaviour.cs:91]: Again, z scale should not be 0.

File: MicrowaveGame/Assets/Resources/Scripts/Items/ItemProjectileSizeBehaviour.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+namespace Scripts.Items
+{
+    public class ItemProjectileSizeBehaviour : ItemBehaviour
+    {
+		private PlayerShootBehaviour _playerShootBehaviour;
+
+		private GameObject _projectilePrefab;
+
+		private Vector3 AddScale = new Vector3(0.05f,0.05f,0);
```
Comment (102107466) [inline MicrowaveGame/Assets/Resources/Scripts/Items/ItemProjectileSizeBehaviour.cs:14]: private should be `_addScale`

**Review by 102107466 — CHANGES_REQUESTED** [review]
Some things I think should be addressed.

**Comment by LachlanMcCaskill** [comment]
The key problem with the projectile size right now is not in this implementation, it's the way the collisions work. We can't make the projectile significantly bigger without causing issues with collisions when the player tries to shoot while next to a wall. Rewriting the collision detection for shooting to accommodate this sounds like too much work, so I would recommend leaving this out and focusing on other remaining tasks.

**Review by 102113528 — APPROVED** [review]
I've lowered the increase size to be more reasonable with four items.

Seems alright otherwise :+1:

**Pull request merged**